app/models/manageiq/providers/openstack/inventory/parser/cloud_manager.rb

```diff
@@ -0,0 +1,426 @@
+        quota.value = value
+        quota.cloud_tenant = persister.cloud_tenants.lazy_find(q["tenant_id"])
+      end
+    end
+  end
+
+  def miq_templates
+    collector.images.each do |i|
+      parent_server_uid = parse_image_parent_id(i)
+      image = persister.miq_templates.find_or_build(i.id)
+      image.type = "ManageIQ::Providers::Openstack::CloudManager::Template"
+      image.uid_ems = i.id
+      image.name = i.name
+      image.vendor = "openstack"
+      image.raw_power_state = "never"
+      image.template = true
+      image.publicly_available = public_image?(i)
+      image.cloud_tenants = image_tenants(i)
+      image.location = "unknown"
+      image.cloud_tenant = persister.cloud_tenants.lazy_find(i.owner) if i.owner
+      image.genealogy_parent = persister.vms.lazy_find(parent_server_uid) unless parent_server_uid.nil?
+
+      hardware = persister.hardwares.find_or_build(i.id)
+      hardware.vm_or_template = image
+      hardware.bitness = image_architecture(i)
+      hardware.disk_size_minimum = (i.min_disk * 1.gigabyte)
+      hardware.memory_mb_minimum = i.min_ram
+      hardware.root_device_type = i.disk_format
+      hardware.size_on_disk = i.size
+      hardware.virtualization_type = i.properties.try(:[], 'hypervisor_type') || i.attributes['hypervisor_type']
+    end
+  end
+
+  def orchestration_stack_resources(stack)
+    raw_resources = collector.orchestration_resources(stack)
+    # reject resources that don't have a physical resource id, because that
+    # means they failed to be successfully created
+    raw_resources.reject! { |r| r.physical_resource_id.nil? }
+    raw_resources.each do |resource|
+      uid = resource.physical_resource_id
+      o = persister.orchestration_stacks_resources.find_or_build(uid)
+      o.ems_ref = uid
+      o.logical_resource = resource.logical_resource_id
+      o.physical_resource = resource.physical_resource_id
+      o.resource_category = resource.resource_type
+      o.resource_status = resource.resource_status
+      o.resource_status_reason = resource.resource_status_reason
+      o.last_updated = resource.updated_time
+      o.stack = persister.orchestration_stacks.lazy_find(stack.id)
+
+      s = persister.vms.find_or_build(uid)
+      s.orchestration_stack = persister.orchestration_stacks.lazy_find(stack.id)
+    end
+  end
+
+  def orchestration_stack_parameters(stack)
+    raw_parameters = collector.orchestration_parameters(stack)
+    raw_parameters.each do |param_key, param_val|
+      uid = compose_ems_ref(stack.id, param_key)
+      o = persister.orchestration_stacks_parameters.find_or_build(uid)
+      o.ems_ref = uid
+      o.name = param_key
+      o.value = param_val
+      o.stack = persister.orchestration_stacks.lazy_find(stack.id)
+    end
+  end
+
+  def orchestration_stack_outputs(stack)
+    raw_outputs = collector.orchestration_outputs(stack)
+    raw_outputs.each do |output|
+      uid = compose_ems_ref(stack.id, output['output_key'])
```

Comment:
you can probably do collector.orchestration_outputs(stack).each, it's -1 line :-)